setup.sh

```diff
@@ -38,16 +19,18 @@ if [ -d venv ]; then
 fi
 
 # Check for correct python version
-VERSION=`python2 -V | awk '{print $2}'`
-if [ "${VERSION:0:1}" -ne "2" ] || [ "${VERSION:2:1}" -ne "7" ]; then
-	     echo "You must use Python 2.7. You are using $VERSION"
-	     return 1
+VERSION=`python3 -V | awk '{print $2}'`
+MAJOR_VERSION=`echo $VERSION | cut -d'.' -f1`  #make sure a python version above 3.7 is used 
+MINOR_VERSION=`echo $VERSION | cut -d'.' -f2`
+if [ $MAJOR_VERSION -eq 3 ] && [ $MINOR_VERSION -ge 7 ]; then
+    echo -e "You are using Python $VERSION"
 else
-	echo -e "You are using Python $VERSION"
+    echo "You must use Python 3.7 or later. You are using $VERSION"
+    return 1
 fi
 
 # Create and activate a clean virtual environment.
-virtualenv --python=python2.7 venv
+virtualenv --python=python3.9 venv
```

Comment:
We should not be using `virtualenv` anymore you should just be able to use `python -m venv`